pkg/adapter/v2/main.go

```diff
@@ -43,7 +43,26 @@ type Adapter interface {
 type AdapterConstructor func(ctx context.Context, env EnvConfigAccessor, client cloudevents.Client) Adapter
 
 func Main(component string, ector EnvConfigConstructor, ctor AdapterConstructor) {
-	MainWithContext(signals.NewContext(), component, ector, ctor)
+
+	ctx := signals.NewContext()
+
+	// TODO(mattmoor): expose a flag that gates this?
+	// ctx = WithHAEnabled(ctx)
```

Comment:
@aslom @lionelvillard we are dropping the use of `enableComponents` for toggling HA in https://github.com/knative/pkg/pull/1476 in favor of:
 * enabling HA by default
 * disabling HA per-component via flag: `--disable-ha` (vs. centralized configmap)

For the adapter HA stuff, I have left things disabled by default, but I also have not connected this toggle to enable HA to anything.  I'm optimistically staging this PR in the hopes that we can devise a plan, so I'd appreciate your thoughts.

Reply:
For adapters, HA should be disabled by default, and that can be done either by:
1. having each adapter explicitly extends the context indicating it supports HA (e.g. `WithHASupport(ctx)`). `--disable-ha` stays the same. 
1.  or by having an `--enable-ha` flag instead of `--disable-ha`.

I prefer 1. since when controllers create adapters they just need to carry around the `--disable-ha` flag down to the adapters, which is nice.

Reply:
Ok, that's essentially what's here.  PTAL and LMK if there are things that should change.

Reply:
I think it's fine, knowing that we need another PR to enable HA for mtping. Do you mind opening an issue tracking this and assign me? Thx!

Reply:
yep, will do.  thanks!

Reply:
https://github.com/knative/eventing/issues/3571